fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java

```diff
@@ -488,19 +498,31 @@ private TOlapTableLocationParam createLocation(OlapTable table) throws UserExcep
 
         // check if disk capacity reach limit
         // this is for load process, so use high water mark to check
-        Status st = GlobalStateMgr.getCurrentSystemInfo().checkExceedDiskCapacityLimit(allBePathsMap, true);
-        if (!st.ok()) {
-            throw new DdlException(st.getErrorMsg());
+        if (!Config.use_staros) {
+            Status st = GlobalStateMgr.getCurrentSystemInfo().checkExceedDiskCapacityLimit(allBePathsMap, true);
+            if (!st.ok()) {
+                throw new DdlException(st.getErrorMsg());
+            }
         }
+
         return locationParam;
     }
 
-    private TNodesInfo createStarrocksNodesInfo() {
+    private TNodesInfo createStarrocksNodesInfo() throws UserException {
         TNodesInfo nodesInfo = new TNodesInfo();
         SystemInfoService systemInfoService = GlobalStateMgr.getCurrentState().getOrCreateSystemInfo(clusterId);
-        for (Long id : systemInfoService.getBackendIds(false)) {
-            Backend backend = systemInfoService.getBackend(id);
-            nodesInfo.addToNodes(new TNodeInfo(backend.getId(), 0, backend.getHost(), backend.getBrpcPort()));
+        try {
+            String currentWarehouseName = ConnectContext.get().getCurrentWarehouse();
```

Comment:
Should use the warehouse in the load job and the warehouse may be specified by user